### DISCUSSION TITLE
为什么vitepress不能容忍<xxx>这样的markdown代码？

### DISCUSSION THREAD

**Discussion by zuoer96** [vuejs/vitepress#2965] [Q&A]
尝试了很多次，比如：

3） 使用 **git add <file>** 命令，添加文件。可以一次添加多个文件。
![image](https://github.com/vuejs/vitepress/assets/31268184/6ca648a6-ef21-4176-a159-d6597e6a9b8c)

这段文本在vitepress中是一定会报错的。
想知道为什么vitepress要关心用户的内容？

**了解到的手段：**
增加闭合标签；用\转义；增加反引号`包裹；
这些手段都可以解决
但是这样做是否会增加用户的负担？
**我在编写markdown的内容时，要额外记忆这件事。**

ps: 仅出于想了解设计理念，不做任何评判。

**Comment by Zhengqbbb** [comment]
这无关用户负担，哪怕 Vitepress 不报错页面也无法显示。 
`<xxx>` 会被认为是 html 标签，这是写 markdown  应该知道的。md 文件本身就可以支持 html 的显示，总要给他一种识别的方法 （是内容还是元素标签）

为什么不使用下面的语法

```md
`git add <file>`
```

https://github.com/vuejs/vitepress/assets/40693636/cfecb5dd-a36c-4829-99c3-a7535d3c5b7e

**Comment by zuoer96** [comment]
> 1. `<xxx>` 会被认为是 html 标签。md 文件本身就可以支持 html 的显示，总要给他一种识别的方法吧
> 2. 不过为什么不使用下面的方法
> 
> ```
> `git add <file>`
> ```
> 
>  CleanShot.2023-09-15.at.11.06.00.mp4


你好，这里的关注点是侧重于：
1.这种阻断式的效果给用户带来的体验，正如上面的例子 **git add <file>** (这里加了`<file>`)。浏览器并没有阻断所有文本显示不是吗？
2.就算用输出`<file>`，这样的代码不显示，可能为什么框架要主动去修正用户的行为？框架是指辅助用户展示内容，为什么要关心用户的内容是否正确？

**Reply by Zhengqbbb** [reply]
是，正常写 Markdown 不会进行阻断，但这是 Vitepress ，需要过一遍生成器渲染器，当发现你的文件中包含标签，但并非正确的标签 (非Html 标签 或是 非组件 还是非闭合的 Html 标签) 时就该退出

这还是得严格，不能宽松，否则页面水合 Vue 就会失败

**Reply by zuoer96** [reply]
补充：死链也会检测，我写个链接，它会检测链接是否有效。
比如笔记内容为：
安装成功后，访问: http://localhost:8081/nexus/
这段话在访问的时候会报错，但它实际上就是我想写的内容
ps：用markdown语法[本地连接](http://localhost:8081/nexus/)，也会被检测

**Comment by hf-xz** [comment]
**既然要享受框架带来的便利，必然要付出代价。**

要只是想单纯的、纯粹的展示 markdown，大可直接用 markdown-it 渲染就好。但话又说回来，为什么写 markdown 还要有这个语法那个语法呢？你在享受 markdown 带来的便利，就遵守了 markdown 给你的规范。对一个框架是同理的。

我的建议是，习惯就好。

**Comment by joisun** [comment]
我这里也出现了这个情况， 不过我是在代码块中出现的：
![image](https://github.com/vuejs/vitepress/assets/44388103/97e2b27c-548f-4c45-bffa-6bfd671225f5)

这有解决办法吗？
